src/main/java/org/traccar/database/DataManager.java

```diff
@@ -328,6 +328,42 @@ public void updateDeviceStatus(Device device) throws SQLException {
                 .executeQuery(Position.class);
     }
 
+    public Position getPositionByTime(long deviceId, Date date) throws SQLException {
+        Collection<Position> positions = QueryBuilder.create(dataSource, getQuery("database.selectPositionByTime"))
+                .setLong("deviceId", deviceId)
+                .setDate("time", date)
+                .executeQuery(Position.class);
+        if (positions.isEmpty()) {
+            return null;
+        } else {
+            return positions.iterator().next();
+        }
+    }
+
+    public Position getPrevPosition(long deviceId, Date date) throws SQLException {
+        Collection<Position> positions = QueryBuilder.create(dataSource, getQuery("database.selectPrevPosition"))
+                .setLong("deviceId", deviceId)
+                .setDate("time", date)
+                .executeQuery(Position.class);
+        if (positions.isEmpty()) {
+            return null;
+        } else {
+            return positions.iterator().next();
+        }
+    }
+
+    public Position getNextPosition(long deviceId, Date date) throws SQLException {
```

Comment:
We only need one of these now, right? Also, please use full words, like `previous` instead of `prev`.

Reply:
Can I rename it everywhere to `preceeding`? It's much clearer title

Reply:
Sounds good